primitives/externalities/src/lib.rs

```diff
@@ -208,7 +245,7 @@ pub trait Externalities: ExtensionStore {
 		&mut self,
 		child_info: &ChildInfo,
 		state_version: StateVersion,
-	) -> Vec<u8>;
+	) -> Option<Vec<u8>>;
```

Comment:
Returning "Option" here is only for the transient storage. Alternatively we could also have transient storage use there own 'root/hash' api.